utils/backoff_test.go

```diff
@@ -0,0 +1,64 @@
+package utils
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func TestProgressiveRetryShouldFail(t *testing.T) {
+	i := 0
+	operationError := errors.New("Operation Failed")
+	retryOperation := func() error {
+		i++
+		return operationError
+	}
+
+	err := ProgressiveRetry(retryOperation)
+
+	if err == nil || err != operationError {
+		t.Errorf("Expected %v, Got %v", operationError, err)
+	}
+	if i != 4 {
+		t.Errorf("Invalid number of attempts: %d", i)
+	}
+}
+
+func TestProgressiveRetryShouldSuccessAfterTwoAttempts(t *testing.T) {
```

Comment:
Perhaps these two tests could be a single, `TestProgressiveRetry` and then have subtests for the two paths?